documentation/library_guide/extension_api.rst

```diff
@@ -139,8 +139,14 @@ header.  All iterators are implemented in the ``oneapi::dpl`` namespace.
   The ``transform_iterator`` is an iterator defined over another iterator whose dereferenced value is the result
   of a function applied to the corresponding element of the original iterator.  Both the type of the original
   iterator and the unary function applied during dereference operations are required template parameters of
-  the ``transform_iterator`` class. The constructor of the ``transform_iterator`` receives both the original
-  iterator and an instance of the unary transform operation as well.
+  the ``transform_iterator`` class. There are three ways to call the constructor of the ``transform_iterator`` class:
```

Comment:
This is good, but I think it's a little verbose with the repeated "Call it passing" and "in this way".  One alternative would be making the list the constructor interface, followed by the description.  Something like the following.

The ``transform_iterator`` class provides three constructors:
* ``transform_iterator()`` instantiates the iterator using default constructed base iterator and unary functor.
* ``transform_iterator(iter)`` instantiates the iterator using the base iterator provided and a default constructed unary functor.
* ``transform_iterator(iter, func)`` instantiates the iterator using the base iterator and unary functor provided.

I don't have a strong opinion on this though.  Either way is fine in my opinion.

Reply:
I like @timmiesmith suggestion and think that we should go with that.

Reply:
I liked it too. Applied it.

Reply:
I liked it too, applied.